### DISCUSSION TITLE
Release document generation failure

### DISCUSSION THREAD

**Discussion by dixonjoel** [ni/measurementlink-python#63] [General]
@mshafer-NI the release step failed trying to push the change to main. Do you know how to get around this?
<img width="880" alt="image" src="https://user-images.githubusercontent.com/38357562/174611979-3dc8d321-8ce9-41ff-a605-7ebdd6dea604.png">

**Comment by mshafer-NI** [comment]
I'll take a look at the release pipeline to see what's different from the ni-python-styleguide repo.

**Comment by mshafer-NI** [comment]
This is the only part that is different that I would expect to be related:
```
---  6/20/2022  9:12:24 AM https://github.com/ni/python-styleguide/blob/main/.github/workflows/Publish-Package.yml
+++  6/20/2022  9:12:24 AM https://github.com/ni/measurement-services-python/blob/main/.github/workflows/Publish.yml
@@ -1,6 +1,2 @@
     steps:
       - uses: actions/checkout@v2
-        with:
-          ref: ${{ github.event.release.target_commitish }} # This is the branch the release was created from. Normally main, but can be a dev branch
-          persist-credentials: false # otherwise, the token used is the GITHUB_TOKEN, instead of your personal token
-          fetch-depth: 0 # otherwise, you will failed to push refs to dest repo
...
```

I can't actually see the branch protection setting on either this repo or on ni-python-styleguide, but these are the settings I'm using on a personal repo that I use in a similar release pipeline on:
![image](https://user-images.githubusercontent.com/23644905/174621978-8b024098-fd25-4170-bc57-685d369875d6.png)

I know that you NEED "include admins" unchecked, but I'm not sure about "require PR" or "require status checks".

Alternatively, this GH-action says it can get around this by temporarily changing policies: https://github.com/marketplace/actions/push-to-status-check-protected-branches#pat-user-permissions - but if opting to use this, make sure to read through the code first and use a specific version: https://github.com/CasperWA/push-protected/tree/main/push_action

**Reply by dixonjoel** [reply]
@Vasantha-Kumar-J is checking on the usage of the Admin PAT, so I believe we're trying the alternate you suggested. I'm not sure if we're using that specific version you mentioned. Vasanth can check on that as well.

**Reply by Vasantha-Kumar-J** [reply]
- In Publish.yml , ADMIN_PAT was used with [ad-m/github-push-action@master](https://github.com/ad-m/github-push-action)
- Now I have updated the Publish.yml to use ADMIN_PAT with [CasperWA/push-protected@v2.10.0](https://github.com/marketplace/actions/push-to-status-check-protected-branches#pat-user-permissions) in this commit :https://github.com/ni/measurement-services-python/commit/c44bf5787e9693ff4440e5d05a6d2af312ddd4a0

After the above changes, I have made test releases and the Publish.yml workflow is running successfully with our repo's custom configuration of branch protection rule where we have "require PR" or "require status checks" enabled.

**Reply by Vasantha-Kumar-J** [reply]
Now the gh-page for our documentation is available for validation : https://symmetrical-journey-236e860a.pages.github.io/

**Reply by Vasantha-Kumar-J** [reply]
FYI, @Tharun-Sundar
